src/v/cloud_storage/recursive_directory_walker.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cloud_storage/recursive_directory_walker.h"
+
+#include "cloud_storage/logger.h"
+#include "utils/gate_guard.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/sstring.hh>
+
+#include <algorithm>
+#include <chrono>
+
+namespace cloud_storage {
+
+ss::lowres_clock::time_point
+recursive_directory_walker::get_timepoint_from_stats(auto stats) {
```

Comment:
I guess this could be a static function (so that the prototype is not in the .h file)

Reply:
this function is already static, what do you mean about prototype is not in the .h file?

Reply:
Yeah sorry for not being clear - I mean standalone static function. If the prototype is not in the .h file, this means we don't need to recompile other translation units if it changes.

Reply:
any way, with using ss::file_stats I don't need this function